pulp_rpm/app/tasks/publishing.py

```diff
@@ -342,7 +348,9 @@ def publish(
     )
     with tempfile.TemporaryDirectory("."):
         with RpmPublication.create(repository_version) as publication:
-            publication.metadata_checksum_type = get_checksum_type("primary", checksum_types)
+            publication.metadata_checksum_type = get_checksum_type(
+                "primary", checksum_types, repo_version=repository_version
+            )
```

Comment:
Could we rewrite it to avoid touching `get_checksum_type()`?

```suggestion
        with RpmPublication.create(repository_version) as publication:
            kwargs = {}
            try:
                first_package = repo_version.content.filter(pulp_type=Package.get_pulp_type()).first()
                kwargs["default"] = first_package.rpm_package.checksum_type
            except WhateverExeptionGoesHere:
                pass
            publication.metadata_checksum_type = get_checksum_type("primary", checksum_types, **kwargs)
```